### PR TITLE
Added 3 new markdown pages for "How-to Start Developing", "How-to Bui…

### PR DESCRIPTION
**Pull request by ChelseaChX**
"How-to Start Developing", "How-to Build Shards" and "How-to Run Shards".

Before submitting your PR, please review the following checklist:

- [x] **DO NOT** submit a PR without having discussed the change first in a related issue. If none exist, create one first.
- [x] **CONSIDER** adding a unit test if your PR resolves an issue.
- [x] **DO** keep pull requests small so they can be easily reviewed.
- [x] **DO** make sure all unit tests pass.
- [x] **DO** make sure not to introduce any new compiler warnings.
- [x] **AVOID** breaking the continuous integration build.
- [x] **AVOID** making significant changes to the overall architecture.

**Description**

"How-to Start Developing"
Reworked an existing tutorial "Contribute → Code → Getting started" which had a nested tutorial under "Contribute → Docs → Getting started". It also had interchangeable use of Linux and Windows commands which can cause errors when following the tutorial, alongside other nested instructions and multiple grammar and spelling errors which can make the following of the tutorial rather difficult. The new tutorial has been streamlined and shifted to the How-to category.

"How-to Build Shards"
Reworked an existing tutorial "Contribute → Code → Building Shards" which can be quite technical and difficult to follow for new users.  Many steps are also done without a clear explanation which can be confusing. The fix for MinGW being unable to read paths added to the user's environment has also been added to the new tutorial. More emphasis is placed on separating the Release and Debug builds as they tend to cause errors when built to the same folder.

"How-to Run Shards"
Added a new tutorial as this was only briefly touched on in the existing "Contribute → Code → Building Shards" tutorial, which might make it difficult to find for users who simply wish to learn how to run Shards instead of building it.

### CODE REVIEW

**Comment by alebaffa** [comment]
Why these guides are different from the Shards website? https://docs.fragcolor.xyz/contribute/code/building-shards/

File: docs/docs/how-to/start-developing.md [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        ```
+
+Input `rustup default nightly` to use the nightly version of Rust. The nightly release is updated more frequently compared to the stable and beta versions.
+	```
+    rustup default nightly
+	```
+
+Input the command `rustup target add wasm32-unknown-unknown --toolchain nightly` to add support for building to web browsers.
+	```
+    rustup target add wasm32-unknown-unknown --toolchain nightly
```
Comment (Kryptos-FR) [inline docs/docs/how-to/start-developing.md:83]: It's correct but rust users are more familiar with this "kind-of shortcut" version:
```suggestion
    rustup +nightly target add wasm32-unknown-unknown
```

**Review by Kryptos-FR — CHANGES_REQUESTED** [review]
Good work!

I added some suggestion and there is a paragraph to fix regarding clojure (we shouldn't mention it).

Also some images are not used (e.g. `docs/docs/how-to/assets/install-mingw-w64-gcc-done.png`). They should be removed from this PR. You can keep them locally for later.

File: docs/docs/how-to/start-developing.md [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+##  Git & GitHub ##
+
+Git is a system used for managing and tracking changes in your code. It is also known as a Version Control System (VCS). It makes it easier for collaborators to work together, and allows you to access our projects too!
+
+Install Git [here](https://git-scm.com/download). The installation settings can be left unchanged.
+
+Although Git is powerful, it can be daunting to use on its own. Instead, we use GitHub, a service which employs Git’s version control with its own to make project collaboration much easier. 
+
+Download GitHub for Desktop [here](https://desktop.github.com/). You can install it with the default settings, and create a GitHub account when prompted if you do not have one.
```
[the remark this answers is not in the document]
Comment (ChelseaChX) [reply docs/docs/how-to/start-developing.md:100]: Would this be better?

```
Although Git is powerful, it can be daunting to use on its own. Instead, we use GitHub, a service which employs Git’s version control with its own to make project collaboration much easier. 

We recommend that you use a [Git GUI Client](https://git-scm.com/downloads/guis) to facilitate your work by pruning off the manual input of code and making the version control process more visual.

We will be using [GitHub Desktop](https://desktop.github.com/) for the tutorials. You can install it with the default settings, and create a GitHub account when prompted if you do not have one.

!!! note
    Even though we recommend GitHub Desktop, feel free to use any GUI client of your choice!
```

File: docs/docs/how-to/build-shards.md [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+---
+
+# Build Shards
+
+!!! note
+    This guide is for new users unfamiliar with the Shards building process. 
+    
+    Click [here](#overview) to skip the tutorial and jump to the overview.
+
+How-to build Shards.
```
[the remark this answers is not in the document]
Comment (ChelseaChX) [reply docs/docs/how-to/build-shards.md:13]: The "How-to" + Page Name was used as the opening line previously so I have left it as such. I think it can be omitted since it is a little redundant. Will do so for the next commit.

File: docs/docs/how-to/run-shards.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+Select the New Folder button ![New Folder Icon](assets/vscode-new-folder-icon.png) near the top of your Explorer window and name the folder “scripts”.
+
+![Create a new “scripts” folder.](assets/vscode-new-folder.png)
+
+Right-click the “scripts” folder and select “New File…”. Name the file “helloworld.edn”.
+
+![Create the “helloworld.edn” file in the “scripts” folder.](assets/vscode-new-file.png)
+
+??? ".edn"
+	EDN is a file format used by Clojure, one of the programming languages employed by Shards.
```
[the remark this answers is not in the document]
Comment (ChelseaChX) [reply docs/docs/how-to/run-shards.md:32]: Would it be alright to rephrase it to "EDN is a file format used by Clojure, a programming language Shards references from"?

If not, I can remove this segment about ".edn" files.

File: docs/docs/how-to/start-developing.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (Kryptos-FR) [reply docs/docs/how-to/start-developing.md:100]: Seems fair enough. I didn't know there was a page with a list of GUIs.

File: docs/docs/how-to/run-shards.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (Kryptos-FR) [reply docs/docs/how-to/run-shards.md:32]: I would remove any mention to Clojure for now to avoid confusion. Or maybe something like: "EDN is a file format used by other programming language such as Clojure"

File: docs/docs/how-to/build-shards.md [hunk 5]
```diff
@@ -68,6 +66,36 @@ If your Shards repository is located at `C:\Fragcolor\Shards`, the command used
...
+
+![Select the “Pull origin” button to check for updates.](assets/github-desktop-pull.png)
+
+The Shards repository contains repositories of other projects, known as *submodules*. When updating your Shards repository, these submodules are left alone and must be manually updated. You can update them with the following command:
+```
+git submodule update --init --recursive
+```
+
+??? help "git: command not found"
+    This error occurs when WinGW is unable to find where git.exe is. We will have to manually add the path of the folder containing it to the MinGW environment.
```
Comment (Kryptos-FR) [inline docs/docs/how-to/build-shards.md:87]: Typo

```suggestion
    This error occurs when MinGW is unable to find where git.exe is. We will have to manually add the path of the folder containing it to the MinGW environment. 
```

File: docs/docs/how-to/build-shards.md [hunk 6]
```diff
@@ -129,18 +157,14 @@ mkdir release
...
-```
-
-Input the command `cmake -G Ninja -DCMAKE_BUILD_TYPE=Debug ../..` to generate the build files.
+Input the following command to generate the build files:
 
 === "Command"
 
     ```
-    cmake -G Ninja -DCMAKE_BUILD_TYPE=Debug ../..
+    cmake -G Ninja -DCMAKE_BUILD_TYPE=Debug -B./Debug ..
```
Comment (Kryptos-FR) [inline docs/docs/how-to/build-shards.md:167]: We could add a quick "note" to explain the `..` part as people might miss it or not understand why it's there.

It is here because the main `CMakeLists.txt` file is at the repository root level (hence one level up from the `build` folder where we run that command) and that's the file `cmake` looks for to determine what to generate.

**Review by Kryptos-FR — CHANGES_REQUESTED** [review]
LGTM.

A typo and a suggestion for a quick note. And then I think we can merge it.

**Review by Kryptos-FR — APPROVED** [review]

**Comment by sinkingsugar** [comment]
How To section of our docs page?
We need to think of a more intuitive structure for docs page btw
